apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -79,3 +444,62 @@ def _from_table(self, arrow_table: pa.Table) -> pa.SparseCOOTensor:
             ]
         ).T
         return pa.SparseCOOTensor.from_numpy(coo_data, coo_coords, shape=self.shape)
+
+
+def _arrow_table_reader(sr: clib.SOMAArray) -> Iterator[pa.Table]:
+    """Private. Simple Table iterator on any Array"""
+    tbl = sr.read_next()
+    while tbl is not None:
+        yield tbl
+        tbl = sr.read_next()
```

Comment:
Idle thought: This seems like it could somehow be `iter(???, None)` but the wrapper code to turn it into that ends up being more than this anyway.

Reply:
yeah, I have played with this a bit, and haven't found anything cleaner/terser.